### PR TITLE
CMake: Miscellaneous cleanup

### PR DESCRIPTION
**Pull request by tamiko**
This pull request is a minor code reshuffling and cleanup:

9e32a3a (Matthias Maier, 79 seconds ago)
   CMake: Remove a dysfunctional workaround

   This does not work.

062080f (Matthias Maier, 3 minutes ago)
   Update comment, fix whitespace

8e82bd9 (Matthias Maier, 4 hours ago)
   CMake: Refactor configuration for MSVC's compiler warning

   Those configuration should go to setup_compiler_flags_msvc.cmake and not
   into the config.h header file...

3e09c34 (Matthias Maier, 4 hours ago)
   CMake: Remove a commented out check

### CODE REVIEW

**Comment by bangerth** [comment]
This is ok, but I'd like to figure out why we disable a whole bunch of warnings that appear to be quite worthwhile to have in https://github.com/tamiko/dealii/commit/8e82bd9eb48f230d691b86f1bcbfa662949b8786

I know that you're just moving around code but let's discuss these at least.

**Comment by tamiko** [comment]
@bangerth I have not investigated the history of this configuration at all. I also do not have any form of output. I'll reboot into a Windows partition tomorrow and we can have a look :-]

Agreed, some of the warnings are quite astonishing.

I'd suggest that we either merge with (a) all warning suppresions in place, (b) with none, (c) something in the middle and figure out the next days what would be a good compromise. I'm slightly in favor of the status quo - i.e., merging as is.

File: cmake/checks/check_02_system_features.cmake [hunk 1]
```diff
@@ -106,51 +106,6 @@ ENDIF()
 ########################################################################
 
 
-IF( CMAKE_SYSTEM_NAME MATCHES "CYGWIN"
```
Comment (tjhei) [inline cmake/checks/check_02_system_features.cmake]: What happens with this change if you configure with cygwin or mingw? Can we just blacklist them with a helpful error message as in "not supported, try MSVC if you are brave"?

Comment (tamiko) [inline cmake/checks/check_02_system_features.cmake]: > What happens with this change if you configure with cygwin or mingw?
> Can we just blacklist them with a helpful error message as in "not
> supported, try MSVC if you are brave"?

We can. What about a separate pull request?

**Pull request merged**